File: backend/src/main/java/com/cognitum/backend/dto/response/ResponseFlashcard.java
package com.cognitum.backend.dto.response;

import com.cognitum.backend.enums.FlashcardType;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class ResponseFlashcard {

    private Long id;
    private String question;
    private String answer;
    private FlashcardType type;
    @JsonProperty("is_retired")
    private Boolean isRetired;
    @JsonProperty("is_stale")
    private Boolean isStale;
    @JsonProperty("easiness_factor")
    private Double easinessFactor;
    private Integer interval;
    private Integer repetitions;
    @JsonProperty("next_review")
    private LocalDate nextReview;
    @JsonProperty("last_reviewed")
    private LocalDateTime lastReviewed;
    @JsonProperty("note_id")
    private Long noteId;

}

File: frontend/src-tauri/src/entities/response_flashcard.rs
use chrono::{DateTime, NaiveDate, Utc};
use serde::{Deserialize, Serialize};
use ts_rs::TS;

#[derive(Serialize, Deserialize, TS)]
#[ts(export)]
pub struct ResponseFlashcard {
    pub id: u64,
    pub question: String,
    pub answer: String,
    pub r#type: String,
    pub is_retired: bool,
    pub is_stale: bool,
    pub easiness_factor: f64,
    pub interval: u32,
    pub repetitions: u32,
    #[ts(type = "string")]
    pub next_review: NaiveDate,
    #[ts(type = "string")]
    pub last_reviewed: Option<DateTime<Utc>>,
    pub note_id: u64,
}
